src/System.Security.Cryptography.X509Certificates/tests/CertificateCreation/PrivateKeyAssociationTests.cs

```diff
@@ -0,0 +1,518 @@
+            {
+                CngKeyCreationParameters creationParameters = new CngKeyCreationParameters()
+                {
+                    ExportPolicy = CngExportPolicies.None,
+                    Provider = CngProvider.MicrosoftSoftwareKeyStorageProvider,
+                    KeyCreationOptions = CngKeyCreationOptions.OverwriteExistingKey,
+                };
+
+                cngKey = CngKey.Create(CngAlgorithm.Rsa, KeyName, creationParameters);
+
+                using (RSACng rsaCng = new RSACng(cngKey))
+                {
+                    CertificateRequest request = new CertificateRequest(
+                        $"CN={KeyName}",
+                        rsaCng,
+                        HashAlgorithmName.SHA256);
+
+                    DateTimeOffset now = DateTimeOffset.UtcNow;
+
+                    using (X509Certificate2 cert = request.CreateSelfSigned(now, now.AddDays(1)))
+                    using (RSA rsa = cert.GetRSAPrivateKey())
+                    {
+                        signature = rsa.SignData(Array.Empty<byte>(), hashAlgorithm, RSASignaturePadding.Pkcs1);
+
+                        Assert.True(
+                            rsaCng.VerifyData(Array.Empty<byte>(), signature, hashAlgorithm, RSASignaturePadding.Pkcs1));
+                    }
+                }
+
+                // Some certs have disposed, did they delete the key?
+                using (CngKey stillPersistedKey = CngKey.Open(KeyName, CngProvider.MicrosoftSoftwareKeyStorageProvider))
+                using (RSACng rsaCng = new RSACng(stillPersistedKey))
+                {
+                    byte[] signature2 = rsaCng.SignData(Array.Empty<byte>(), hashAlgorithm, RSASignaturePadding.Pkcs1);
+
+                    Assert.Equal(signature, signature2);
+                }
+            }
+            finally
+            {
+                cngKey?.Delete();
+            }
+        }
+
+        [Fact]
+        public static void ThirdPartyProvider_RSA()
+        {
+            using (RSA rsaOther = new RSAOther())
+            {
+                HashAlgorithmName hashAlgorithm = HashAlgorithmName.SHA256;
+
+                CertificateRequest request = new CertificateRequest(
+                    $"CN={nameof(ThirdPartyProvider_RSA)}",
+                    rsaOther,
+                    hashAlgorithm);
+
+                byte[] signature;
+                byte[] data = request.SubjectName.RawData;
+
+                DateTimeOffset now = DateTimeOffset.UtcNow;
+
+                using (X509Certificate2 cert = request.CreateSelfSigned(now, now.AddDays(1)))
+                using (RSA rsa = cert.GetRSAPrivateKey())
+                {
+                    signature = rsa.SignData(data, hashAlgorithm, RSASignaturePadding.Pkcs1);
+                }
+
+                Assert.True(rsaOther.VerifyData(data, signature, hashAlgorithm, RSASignaturePadding.Pkcs1));
+            }
+        }
+
+        [Theory]
+        [PlatformSpecific(TestPlatforms.Windows)]
```

Comment:
Not via CspParameters, though...  this test is all about persistent storage keys.